lib/ansible/module_utils/common/validation.py

```diff
@@ -506,17 +506,12 @@ def check_type_int(value):
 
     :return: int of given value
     """
-    if isinstance(value, integer_types):
-        return value
-
-    if isinstance(value, string_types):
-        try:
-            return int(value)
-        except ValueError:
-            pass
-
-    raise TypeError('%s cannot be converted to an int' % type(value))
-
+    if not isinstance(value, int):
```

Comment:
Why is this function being changed? The existing behavior strictly requires an int, so there's no risk of truncation. The new logic is more relaxed, allowing values it previously did not accept.

Reply:
i was making it consistent with the rest, it will allow things that are 'cleanly' castable to int as do the other 2